pkg/resources/cloudcontroller/kubevirt.go

```diff
@@ -110,9 +98,22 @@ func kubevirtDeploymentCreator(data *resources.TemplateData) reconciling.NamedDe
 			}
 
 			defResourceRequirements := map[string]*corev1.ResourceRequirements{
-				ccmContainerName:    kvResourceRequirements.DeepCopy(),
-				openvpnSidecar.Name: openvpnSidecar.Resources.DeepCopy(),
+				ccmContainerName: kvResourceRequirements.DeepCopy(),
 			}
+
+			if !data.IsKonnectivityEnabled() {
```

Comment:
that can bring issues to user cluster if both k8s clusters (KubeVirt and user cluster) uses node local dns (a loop problem). Why if konnectivity is enabled it's not allowed to change DNS policy and add custom servers?

Reply:
Not sure if I am getting the problem with the loop, but maybe it will help if I describe the rationale behind this change:

Before this change, the cloud controller pod in seed (as well as some other control plane components running in the seed) had the DNS policy set to `None` and instead of using the default k8s DNS of the seed cluster, it was configured with a custom DNS resolver running for the user cluster in the seed. I was under the impression that the sole role of this custom DNS resolver in seed was to resolve the user cluster services to pod IPs when control plane components (mainly apiserver) needs to contact them (e.g. extension apiservers, webhooks). 

As this functionality is now built in Konenctivity for the apiserver, so the custom dns resolver in the seed for each user cluster is not necessary, at least for this purpose.

But I was not aware of any other role of this dns-resolver, e.g. preventing loops in case of KubeVirt as you are mentioning. If it is really needed, maybe we could still deploy it for KubeVirt user clusters only?

Reply:
The DNS change will be done as a separate change: https://github.com/kubermatic/kubermatic/issues/8256